### PR TITLE
BUG: Fix range_window for base_sql backend (continued)

### LINKED ISSUE #2608: BUG: Range window with order by breaks for base_sql backend
What's wrong
=====
When compiling a range_window in backend based on `base_sql` (BigQuery in the example here), an exception will be thrown as:

```
in /ibis/backends/base_sql/__init__.py in format_window(translator, op, window)
--> 833  if window.max_lookback is not None:

AttributeError: 'NoneType' object has no attribute 'max_lookback'
```
It turns out to be `window` is a `None` object. 
By following the stacktrace, in https://github.com/ibis-project/ibis/blob/master/ibis/backends/base_sql/__init__.py#L821 function `time_range_to_range_window` doesn't have a return value but it is referenced here: https://github.com/ibis-project/ibis/blob/master/ibis/backends/base_sql/__init__.py#L981 as ` window = time_range_to_range_window(translator, window)`

Seems a `window` object should be returned in ` time_range_to_range_window`.

Code to Reproduce
=====
```
import datetime
import ibis
from ibis.expr.types import TableExpr
import ibis.expr.operations as ops

def _get_ibis_client():
    ibis_client = ibis.bigquery.BigQueryClient
    setattr(ibis_client, "__init__", lambda t: None)
    ibis_client = ibis_client()
    return ibis_client

def _get_test_table(ibis_client):
    names = ['Class',                 
  'end_of_period',         
  'M_Total_Dollars_Total',
            ]
    types = ['string', "date", 'float64']
    schema = ibis.Schema(names, types)
    t = TableExpr(ops.SQLQueryResult("select * from t", schema, ibis_client))
    return t

def test_window_compile():
    ibis_client = _get_ibis_client()
    t = _get_test_table(ibis_client)
    key_cols = ['Class', 'end_of_period']
    order_col = 'end_of_period'
    window = ibis.range_window(
        preceding=ibis.interval(days=62),
        order_by=t[order_col],
        group_by=[t['Class']],
    )
    t = t.mutate(window=t["M_Total_Dollars_Total"].sum().over(window))
    t.compile()
    return t

test_window_compile()
```

### PR DESCRIPTION
**Pull request by timothydijamco**
closes #2608

This PR is a duplicate of #2614 (includes @LeeTZ 's bug fix and new test) + additional changes from me to debug the new test failing on some backends. (Creating a new PR just to make it easier for me to iterate and check if CI passes).

**Please see #2614 for a full description of the original problem / solution**

### CODE REVIEW

**Comment by timothydijamco** [comment]
The CI is passing now

The pre-merge CI does not test BigQuery, but it does hit the relevant code when it tests the other SQL backends. I also confirmed locally that the current state of the PR resolves the error with the BigQuery backend mentioned in #2608.

File: ibis/backends/base_sql/__init__.py [hunk 1]
```diff
@@ -733,6 +734,75 @@ def from_unixtime(translator, expr):
...
+    # Literal(1, Interval(value_type=int8, unit='D', nullable=True))
+    # Parse both args and multipy 1 * _map_interval_to_microseconds['D']
+    ops.Literal: mul,
+    ops.IntervalMultiply: mul,
+    ops.IntervalAdd: add,
+    ops.IntervalSubtract: sub,
+}
+
+
+def _replace_interval_with_scalar(expr):
```
Comment (jreback) [inline ibis/backends/base_sql/__init__.py:760]: can you type in and out

**Review by jreback — CHANGES_REQUESTED** [review]
looks good. can you add a whatsnew note (ref the original issue). ping on green.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @timothydijamco and @LeeTZ

**Pull request merged**